File: bancosecha/bancosecha/controllers/customer.py
import frappe
import re

def validate(doc, method):
    validate_phone_number(doc)

def validate_phone_number(doc):
    if doc.custom_phone_number:
        # Remove spaces, dashes, parentheses, and plus signs
        cleaned_number = re.sub(r'[\s\-()+]', '', doc.custom_phone_number)
        if cleaned_number != doc.custom_phone_number:
            doc.custom_phone_number = cleaned_number

File: bancosecha/public/js/customer.js
frappe.ui.form.on("Customer", {
    custom_phone_number(frm){
        console.log("Cleaning phone number")
        if(!frm.doc.custom_phone_number)
            return;
        // Let's clean any space, dashes or parentheses or plus from the phone number
        let cleaned_number = frm.doc.custom_phone_number.replace(/[\s\-()+]/g, '');
        if (frm.doc.custom_phone_number !== cleaned_number) {
            frm.set_value("custom_phone_number", cleaned_number);
        }
    },
});